### ISSUE TITLE
Incompatibility with Qwen

### ISSUE THREAD

**Issue by Ricardokevins** [OpenLLMAI/OpenRLHF#303]
When using PPO for training, I noticed that within the actor's generate function, it locates the position of the EOS token and sets all subsequent tokens to the EOS token. However, this results in even when the padding side is set to left, the far right end of the sequence still contains padding tokens or the EOS token. This has triggered an internal assert in the Qwen model.

```
        if attention_mask is not None and self._attn_implementation == "flash_attention_2" and use_cache:
            is_padding_right = attention_mask[:, -1].sum().item() != batch_size
            if is_padding_right:
                raise ValueError(
                    "You are attempting to perform batched generation with padding_side='right'"
                    " this may lead to unexpected behaviour for Flash Attention version of Qwen2. Make sure to "
                    " call `tokenizer.padding_side  = 'left'` before tokenizing the input. "
                )
```

**Comment by hijkzzz** [comment]
just disable flash_attention_2

**Comment by Ricardokevins** [comment]
> just disable flash_attention_2

Yes, but it may slow down the training

Luckly, Llama3 works fine
